### PR TITLE
LAB-1166 New flags for tools

### PR DESCRIPTION
**Pull request by supraja-968**
## What type of PR is this? 
_Remove the categories that do not apply_

- 🎮 Feature
- 🔋 Optimization

## Description

With this feature, the tool table is modified with 3 new columns:
1. Display yes/no
to decide if the tool need to be shown in the dropdown selection (example: all the test, and previous version of tools can be set to no)
for models having many versions, some can become outdated after a point, so while we might still want to show more than 1 version of a model, we can set this flag to no to not show them in the UI anymore (might have to think this through on how to handle API submissions not through UI)

2. task category
which model belongs to which task - so in the experiment submission, when clicked on a task, it should only show associated models

3. default yes/no
when clicked on a task, 1 model should be auto selected by default (example: colabdesign-gateway-v0.5 is now the default when clicked on a task, though it is hardcoded, we can remove this hard coding, once this is implemented)

## Related Tickets & Documents

Changes to the way tools are added/displayed going forward:

1. The /tools listhandler will display only tools with the display flag set to true by default going forward. Internally, you can display the hidden tools by querying the DB or calling /tools?display=false.
2. When adding a new tool, you can either set it to a task category in the toolJson. If not, it will get added to "other models" task category by default. Labsay and other test version models can be found here.
3. Tools will only get added with display flag yes and default tool flag no from the front end. Internally we can change the default tool for each category and hide tools in the DB.

Note to the team: After deploying this feature, we can cleanup the tools table in prod to have the task category set.

## Steps to Test

_A list of steps a reviewer can walk through to test the changes._

## Relevant GIF 

_Required 🐔_

### CODE REVIEW

File: frontend/lib/redux/slices/toolDetailSlice/slice.ts [hunk 1]
```diff
@@ -50,9 +52,10 @@ export const toolDetailSlice = createSlice({
     setToolDetailSuccess: (state, action: PayloadAction<boolean>) => {
       state.success = action.payload;
     },
+    resetToolDetail: () => { return initialState; },
```
Comment (thetechnocrat-dev) [inline frontend/lib/redux/slices/toolDetailSlice/slice.ts:55]: 💯

**Review by thetechnocrat-dev — CHANGES_REQUESTED** [review]
Code looks good! Once CI tests pass I'll give it an approval to merge (https://github.com/labdao/plex/actions/runs/7762303368/job/21172496979?pr=870#step:11:955).

If you `npm run build` locally from the frontend you will get the production type checking to help you fix the CI build issue.

Ping me if you have questions.

**Review by acashmoney — DISMISSED** [review]
This is looking great @supraja-968 . Can you also include the updates to the OpenAPI spec for these new Tool flags? Don't worry about modifying the entire document, we only need to update the `GET` requests for `/tools` and the `Tool` schema

**Review by thetechnocrat-dev — APPROVED** [review]
💯 on fixing the tests

**Comment by supraja-968** [comment]
> This is looking great @supraja-968 . Can you also include the updates to the OpenAPI spec for these new Tool flags? Don't worry about modifying the entire document, we only need to update the `GET` requests for `/tools` and the `Tool` schema

Done ✅

**Comment by acashmoney** [comment]
> > This is looking great @supraja-968 . Can you also include the updates to the OpenAPI spec for these new Tool flags? Don't worry about modifying the entire document, we only need to update the `GET` requests for `/tools` and the `Tool` schema
> 
> Done ✅

Looks great! Approving

**Pull request merged**